### DISCUSSION TITLE
Autotext component usage to get total page count for autolayout birt report

### DISCUSSION THREAD

**Discussion by Sanket-Kuldharan** [eclipse-birt/birt#1749] [Q&A]
How to get the total page count of autolayout birt report  and store it in persistent global variable to used it for future used?
I tried for the same using on render script but total page type autotext component shows value like page number.
**Important note:** Inbuilt variable 'totalPage' shows incorrect value for autolayout type birt report.community please give me ideas regarding this.

**Comment by speckyspooky** [comment]
Please tae a look to discussion #1799 there I documented the options to handle "pageNumber" and "totalPage" - incl. demo-report.
You can use both values with "onRender" and you cann store it if needed into persistent global variables.
If you need an example of global persistent variable I added an exmaple on discussion #1848.

So you will find all details.
Please be aware the AutoLayout tries to calculate all dynamically and so sometimes the page numbering isn't correct in all cases.
Therfore the recommedation is to use the fixed layout.

**Reply by Sanket-Kuldharan** [reply]
Actually I want the help for autolayout type report.In this type of report ,'totalPage' value shows wrong when we try to display it.it behave same like 'pageNumber' variable.Please guide me regarding this problem.How can we get correct value of 'totalPage' for autolayout type birt report?

**Reply by Sanket-Kuldharan** [reply]
@speckyspooky any idea?please share with me.I have stuck due to this issue

**Reply by Sanket-Kuldharan** [reply]
@speckyspooky Can you please help me to clear my doubt?

**Reply by speckyspooky** [reply]
If you can provide a small test report I can take a look into it.
But the auto layout is very special therefore I work on the most cases with the fixed layout.

**Comment by hvbtup** [comment]
I think that the intention for the auto-layout option was to be able to create HTML pages where there is no physical page size limit, and thus the whole idea of page numbers dos not fit to the auto-layout option IMHO.
